pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -513,3 +536,71 @@ func (r *ReconcileClusterPool) createCloudBuilder(pool *hivev1.ClusterPool, logg
 		return nil, errors.New("unsupported platform")
 	}
 }
+
+// getAllPendingClusterClaims returns all of the ClusterClaims that are requesting clusters from the specified pool.
+// The claims are returned in order of creation time, from oldest to youngest.
+func (r *ReconcileClusterPool) getAllPendingClusterClaims(pool *hivev1.ClusterPool, logger log.FieldLogger) ([]*hivev1.ClusterClaim, error) {
+	claimsList := &hivev1.ClusterClaimList{}
+	if err := r.Client.List(context.Background(), claimsList, client.InNamespace(pool.Namespace)); err != nil {
+		logger.WithError(err).Error("error listing ClusterClaims")
+		return nil, err
+	}
+	var pendingClaims []*hivev1.ClusterClaim
+	for i, claim := range claimsList.Items {
+		// skip claims for other pools
+		if claim.Spec.ClusterPoolName != pool.Name {
+			continue
+		}
+		// skip claims that have been assigned already
+		if claim.Status.Namespace != "" {
+			continue
+		}
+		pendingClaims = append(pendingClaims, &claimsList.Items[i])
+	}
+	sort.Slice(
+		pendingClaims,
+		func(i, j int) bool {
+			return pendingClaims[i].CreationTimestamp.Before(&pendingClaims[j].CreationTimestamp)
+		},
+	)
+	return pendingClaims, nil
+}
+
+func (r *ReconcileClusterPool) assignClustersToClaims(claims []*hivev1.ClusterClaim, cds []*hivev1.ClusterDeployment, logger log.FieldLogger) ([]*hivev1.ClusterDeployment, error) {
+	for _, claim := range claims {
+		var conds []hivev1.ClusterClaimCondition
+		var changed bool
+		if len(cds) > 0 {
+			claim.Status.Namespace = cds[0].Namespace
+			cds = cds[1:]
+			logger.WithField("cluster", claim.Status.Namespace).Info("assigning cluster to claim")
+			conds = controllerutils.SetClusterClaimCondition(
+				claim.Status.Conditions,
+				hivev1.ClusterClaimNotReadyCondition,
+				corev1.ConditionTrue,
+				"ClusterAssigned",
+				"Cluster assigned to claim",
```

Comment:
We're still NotReady=true here, could we expand the message a little to indicate we're waiting for the link to be completed, maybe "Cluster identified for claim, awaiting assignment" or something like that.